# Import important Liberaries and Data

In [39]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
from scipy.stats import describe
import seaborn as sns
import statistics

from sklearn.preprocessing import StandardScaler, minmax_scale


import warnings
warnings.filterwarnings('ignore')

In [2]:
data = pd.read_csv(r"F:\Desktop\ML DSA\DataScience\FeatureEngineering\Obesity Data\Data\train.csv")
train = data.drop('id',axis=1)
data = pd.read_csv(r"F:\Desktop\ML DSA\DataScience\FeatureEngineering\Obesity Data\Data\test.csv")
test = data.drop('id',axis=1)
Origin = pd.read_csv(r"F:\Desktop\ML DSA\DataScience\FeatureEngineering\Obesity Data\Data\Origin_Data.csv")

In [3]:
print(f"Train data have {train.shape[1]} columns and {train.shape[0]} rows")
print(f"Origin data have {Origin.shape[1]} columns and {Origin.shape[0]} rows")
print(f"Test data have {test.shape[1]} columns and {test.shape[0]} rows")

Train data have 17 columns and 20758 rows
Origin data have 17 columns and 2111 rows
Test data have 16 columns and 13840 rows


In [4]:
# Lets concart train and origin data
df = pd.concat([train,Origin])
df.shape

(22869, 17)

## Quick overview of Data

In [5]:
df.head(10)

Gender        Age    Height      Weight family_history_with_overweight  \
0    Male  24.443011  1.699998   81.669950                            yes   
1  Female  18.000000  1.560000   57.000000                            yes   
2  Female  18.000000  1.711460   50.165754                            yes   
3  Female  20.952737  1.710730  131.274851                            yes   
4    Male  31.641081  1.914186   93.798055                            yes   
5    Male  18.128249  1.748524   51.552595                            yes   
6    Male  29.883021  1.754711  112.725005                            yes   
7    Male  29.891473  1.750150  118.206565                            yes   
8    Male  17.000000  1.700000   70.000000                             no   
9  Female  26.000000  1.638836  111.275646                            yes   

  FAVC      FCVC       NCP        CAEC SMOKE      CH2O  SCC       FAF  \
0  yes  2.000000  2.983297   Sometimes    no  2.763573   no  0.000000   
1  yes  2.000000  3.000000  Frequently    no  2.000000   no  1.000000   
2  yes  1.880534  1.411685   Sometimes    no  1.910378   no  0.866045   
3  yes  3.000000  3.000000   Sometimes    no  1.674061   no  1.467863   
4  yes  2.679664  1.971472   Sometimes    no  1.979848   no  1.967973   
5  yes  2.919751  3.000000   Sometimes    no  2.137550   no  1.930033   
6  yes  1.991240  3.000000   Sometimes    no  2.000000   no  0.000000   
7  yes  1.397468  3.000000   Sometimes    no  2.000000   no  0.598655   
8  yes  2.000000  3.000000   Sometimes    no  3.000000  yes  1.000000   
9  yes  3.000000  3.000000   Sometimes    no  2.632253   no  0.000000   

        TUE       CALC                 MTRANS           NObeyesdad  
0  0.976473  Sometimes  Public_Transportation  Overweight_Level_II  
1  1.000000         no             Automobile        Normal_Weight  
2  1.673584         no  Public_Transportation  Insufficient_Weight  
3  0.780199  Sometimes  Public_Transportation     Obesity_Type_III  
4  0.931721  Sometimes  Public_Transportation  Overweight_Level_II  
5  1.000000  Sometimes  Public_Transportation  Insufficient_Weight  
6  0.696948  Sometimes             Automobile      Obesity_Type_II  
7  0.000000  Sometimes             Automobile      Obesity_Type_II  
8  1.000000         no  Public_Transportation   Overweight_Level_I  
9  0.218645  Sometimes  Public_Transportation     Obesity_Type_III

In [6]:
test.head(10)

Gender        Age    Height      Weight family_history_with_overweight  \
0    Male  26.899886  1.848294  120.644178                            yes   
1  Female  21.000000  1.600000   66.000000                            yes   
2  Female  26.000000  1.643355  111.600553                            yes   
3    Male  20.979254  1.553127  103.669116                            yes   
4  Female  26.000000  1.627396  104.835346                            yes   
5    Male  19.799054  1.844751   59.605028                            yes   
6    Male  18.000000  1.728834   51.442293                            yes   
7    Male  20.000000  1.810000   80.000000                            yes   
8    Male  21.000000  1.700000   80.000000                            yes   
9  Female  21.000000  1.560000   53.000000                             no   

  FAVC      FCVC       NCP        CAEC SMOKE      CH2O SCC       FAF  \
0  yes  2.938616  3.000000   Sometimes    no  2.825629  no  0.855400   
1  yes  2.000000  1.000000   Sometimes    no  3.000000  no  1.000000   
2  yes  3.000000  3.000000   Sometimes    no  2.621877  no  0.000000   
3  yes  2.000000  2.977909   Sometimes    no  2.786417  no  0.094851   
4  yes  3.000000  3.000000   Sometimes    no  2.653531  no  0.000000   
5  yes  2.000000  4.000000   Sometimes    no  2.722063  no  2.000000   
6  yes  1.202075  3.000000   Sometimes    no  1.087166  no  0.788585   
7  yes  2.000000  3.000000      Always    no  2.000000  no  3.000000   
8  yes  2.000000  3.000000  Frequently   yes  2.000000  no  0.000000   
9  yes  2.000000  3.000000   Sometimes    no  2.000000  no  0.000000   

        TUE       CALC                 MTRANS  
0  0.000000  Sometimes  Public_Transportation  
1  0.000000  Sometimes  Public_Transportation  
2  0.250502  Sometimes  Public_Transportation  
3  0.000000  Sometimes  Public_Transportation  
4  0.741069  Sometimes  Public_Transportation  
5  1.283673         no             Automobile  
6  1.000000  Sometimes  Public_Transportation  
7  0.000000         no             Automobile  
8  2.000000         no  Public_Transportation  
9  1.000000  Sometimes  Public_Transportation

## 1.1 Summary of Data

In [41]:
df.describe().round(2).style.format(precision=2).background_gradient(
    vmax=75, cmap="Blues"
)

Acording to this sample of data
- Ranges 
    1. Age: From 14 to 61
    2. Height: From 1.45 to 1.98
    3. Weight: From 39 to 173
    4. FCVC: From 1 to 3
    5. NCP: From 1 to 4
    6. CH20: From 1 to 3 
    7. FAF: From 0 to 3
    8. TUE: From 0 to 2

### 1.2 Missing Values

In [8]:
print(df.isnull().sum())

Gender                            0
Age                               0
Height                            0
Weight                            0
family_history_with_overweight    0
FAVC                              0
FCVC                              0
NCP                               0
CAEC                              0
SMOKE                             0
CH2O                              0
SCC                               0
FAF                               0
TUE                               0
CALC                              0
MTRANS                            0
NObeyesdad                        0
dtype: int64


### Insights
1. We have total 17 Columns 
2. Out of 17 Columns, 8 are float values and 9 columns are object type
3. We don't have any missing values in our data

# Exploratry Data Analysis 